watertap/examples/flowsheets/ion_exchange/ion_exchange_demo.py

```diff
@@ -0,0 +1,333 @@
+    ix.resin_bulk_dens.fix()
+    ix.bed_porosity.fix()
+    ix.dimensionless_time.fix()
+    ix.regen_dose.fix()
+    ix.regen_recycle.fix()
+    ix.t_regen.fix()
+    ix.t_bw.fix()
+    ix.bw_rate.fix()
+    ix.rinse_bv.fix()
+    ix.pump_efficiency.fix()
+    ix.p_drop_A.fix()
+    ix.p_drop_B.fix()
+    ix.p_drop_C.fix()
+    ix.bed_expansion_frac_A.fix()
+    ix.bed_expansion_frac_B.fix()
+    ix.bed_expansion_frac_C.fix()
+    ix.service_to_regen_flow_ratio.fix()
+    ix.number_columns_redund.fix()
+
+
+def initialize_system(m):
+    target_ion = m.fs.ion_exchange.config.target_ion
+    m.fs.feed.initialize()
+    m.fs.feed.properties[0].flow_vol_phase["Liq"].fix()
+    m.fs.feed.properties[0].conc_equiv_phase_comp["Liq", target_ion].fix()
+    propagate_state(m.fs.feed_to_ix)
+    m.fs.ion_exchange.initialize()
+    propagate_state(m.fs.ix_to_regen)
+    propagate_state(m.fs.ix_to_product)
+    m.fs.regen.initialize()
+    m.fs.product.initialize()
+    m.fs.costing.initialize()
+
+
+def optimize_system(m):
+    # Example of optimizing number of IX columns based on desired effluent equivalent concentration
+    m.fs.obj = Objective(expr=m.fs.costing.LCOW)
+    ix = m.fs.ion_exchange
+    target_ion = m.fs.ion_exchange.config.target_ion
+    ix.properties_out[0].conc_equiv_phase_comp["Liq", target_ion].fix(0.5)
+    propagate_state(m.fs.ix_to_product)
+    propagate_state(m.fs.ix_to_regen)
+    m.fs.product.initialize()
+    m.fs.regen.initialize()
+
+    # To adjust solution to fixed-pattern to achieve desired effluent, must unfix dimensionless_time
+    ix.dimensionless_time.unfix()
+    # Can optimize around different design variables, e.g., bed_depth, service_flow_rate (or combinations of these)
+    # Here demonstrates optimization around column design
+    ix.number_columns.unfix()
+    ix.bed_depth.unfix()
+    solver.solve(ix)
+    # num_col = np.ceil(ix.number_columns())  # To eliminate fractional number of columns
+    # bed_depth = ix.bed_depth()
+    # ix.bed_depth.fix(bed_depth)
+    # ix.number_columns.fix(num_col)
+
+
+def get_ion_config(ions):
+
+    if not isinstance(ions, list):
+        ions = [ions]
+    diff_data = {
+        "Na_+": 1.33e-9,
+        "Ca_2+": 9.2e-10,
+        "Cl_-": 2.03e-9,
+        "Mg_2+": 0.706e-9,
+        "SO4_2-": 1.06e-9,
+    }
+    mw_data = {
+        "Na_+": 23e-3,
+        "Ca_2+": 40e-3,
+        "Cl_-": 35e-3,
+        "Mg_2+": 24e-3,
+        "SO4_2-": 96e-3,
+    }
+    charge_data = {"Na_+": 1, "Ca_2+": 2, "Cl_-": -1, "Mg_2+": 2, "SO4_2-": -2}
+    ion_config = {
+        "solute_list": [],
+        "diffusivity_data": {},
+        "mw_data": {"H2O": 18e-3},
+        "charge": {},
+    }
+    for ion in ions:
+        ion_config["solute_list"].append(ion)
+        ion_config["diffusivity_data"][("Liq", ion)] = diff_data[ion]
+        ion_config["mw_data"][ion] = mw_data[ion]
+        ion_config["charge"][ion] = charge_data[ion]
+    return ion_config
+
+
+def display_results(m):
+
```

Comment:
How different is this than the `.report()`? If this truly is to serve as a demo what is the value in custom reporting if there is a report options internal to WaterTAP and the IX unit model?

Reply:
We could check what the ion exchange unit model has in its _get_performance_contents method to use `.report()`. With that said, we have been using `display_results` across most flowsheets just to display flowsheet level results that might go beyond a particular unit model. However, we probably could use `.report()` to display the IX model's outputs under this function.

Reply:
IX unit model has both of those methods, but they need to be updated. Will be quick, but I will have to make a separate PR for it. 

Also a downside to ``.report()`` is it doesn't display costing data, which I wanted to include in the demo here.